tests/runtime/builtin

```diff
@@ -242,3 +242,17 @@ NAME kstack builtin with stack_mode config
 RUN {{BPFTRACE}} -e 'config = { stack_mode=raw } k:do_nanosleep { @c[kstack] = 1; exit(); }'
 EXPECT_REGEX ^@c\[\n[0-9a-f]+$
 AFTER ./testprogs/syscall nanosleep  1e8
+
+NAME usermode_builtin_in_kprobe_fentry
+PROG k:vfs_read { printf("%d ", usermode); } fentry:vfs_open { print(usermode); }
+EXPECT 0 0
+AFTER ./testprogs/syscall read
+
+NAME usermode_builtin_in_uprobe
+PROG uprobe:./testprogs/uprobe_symres:test { print(usermode); exit(); }
+EXPECT 1
+AFTER ./testprogs/uprobe_symres
+
+NAME usermode_builtin_in_BEGIN_END
+PROG BEGIN { printf("%d ", usermode); exit(); } END { print(usermode); }
+EXPECT 1 1
```

Comment:
Maybe good to also add a test for other archs to test that usermode is always false.

Reply:
Also needs an "ARCH" directive to prevent this test from running on unsupported architectures. e.g. https://github.com/bpftrace/bpftrace/blob/a8955af9335ea696e29d02d8d3fd8df0c5d2dade/tests/runtime/probe#L191